index.js

```diff
@@ -7,4 +13,62 @@ const fgMultipleHooks = async (req, res, ...multipleHooks) => {
     return shouldAbort;
 };
 
-module.exports = fgMultipleHooks;
\ No newline at end of file
+const defaultLastOnResponse = async (req, res, stream) => {
+    const chunked = stream.headers[TRANSFER_ENCODING_HEADER_NAME]
+        ? stream.headers[TRANSFER_ENCODING_HEADER_NAME].endsWith('chunked')
+        : false
+
+    if (req.headers.connection === 'close' && chunked) {
+        try {
+            // remove transfer-encoding header
+            const transferEncoding = stream.headers[TRANSFER_ENCODING_HEADER_NAME].replace(/(,( )?)?chunked/, '')
+            if (transferEncoding) {
+                // header format includes many encodings, example: gzip, chunked
+                res.setHeader(TRANSFER_ENCODING_HEADER_NAME, transferEncoding)
+            } else {
+                res.removeHeader(TRANSFER_ENCODING_HEADER_NAME)
+            }
+
+            if (!stream.headers['content-length']) {
+                // pack all pieces into 1 buffer to calculate content length
+                const resBuffer = Buffer.concat(await toArray(stream))
+
+                // add content-length header and send the merged response buffer
+                res.setHeader('content-length', '' + Buffer.byteLength(resBuffer))
+                res.statusCode = stream.statusCode
+                res.end(resBuffer)
+            }
+        } catch (err) {
+            res.statusCode = 500
+            res.end(err.message)
+        }
+    } else {
+        res.statusCode = stream.statusCode
+        pump(stream, res)
+    }
+}
+
+const onResponseHooks = async (req, res, stream, last, ...multipleHooks) => {
+    let shouldAbort = false;
+    const promises = multipleHooks.map(async oneHook => {
+        shouldAbort = shouldAbort || await oneHook(req, res);
```

Comment:
every hook should receive the the stream

Reply:
that is particularly what i have been avoiding, any operation on the stream can and should be done in the last hook.

Reply:
on second thoughts, it is still the developer's consumer's responsibility, I'll do the need full changes to allow stream in every Hook